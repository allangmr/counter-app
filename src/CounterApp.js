import {useState} from 'react'
import Proptypes from 'prop-types'

const CounterApp = ({value = 10}) =>{
    const [counter, setCounter] = useState(value);
    //handelAdd
    const handleAdd = () => {
        setCounter(counter+1);
        // setCounter((c)=> c+1 );
    }
    //handleSubtract
    const handleSubtract = () =>  setCounter(counter-1);

    //handleSubtract
    const handleReset = () => setCounter(value);

    return(
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubtract}>-1</button>
            <button onClick={handleReset}>Reset</button>
        </>
    )

}

CounterApp.propTypes = {
    value: Proptypes.number
}

export default CounterApp

