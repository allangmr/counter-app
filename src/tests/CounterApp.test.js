import CounterApp from "../CounterApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'


describe('Test of CounterApp.js', () => {
    let wrapper = shallow(<CounterApp  />);

    beforeEach(() =>{
        wrapper = shallow(<CounterApp  />);
    });


    test('should be show <CounterApp />', () => {

        expect(wrapper).toMatchSnapshot();
    })

    test('should be show <CounterApp /> with default value', () => {
        const wrapper = shallow(<CounterApp />);

        expect(wrapper).toMatchSnapshot();
    })

    test('should be show default value 100 and use wrapper.find', () => {
        const valor = 100;
        const wrapper = shallow(<CounterApp value={valor} />);
        const defValue = wrapper.find('h2').text().trim();
        expect(valor).toBe(Number(defValue));
    })

    test('should be increment counter -1', () => {
        wrapper.find('button').at(1).simulate('click');
        const defValue = wrapper.find('h2').text().trim();

        expect(defValue).toBe('9');
    })

    test('should be dicrement counter +1', () => {
        wrapper.find('button').at(0).simulate('click');
        const defValue = wrapper.find('h2').text().trim();

        expect(defValue).toBe('11');
    })

    test('should be reset counter', () => {
        const wrapper = shallow(<CounterApp value={500} />);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        const defValue = wrapper.find('h2').text().trim();

        expect(defValue).toBe('500');
    })

    
    
    
});