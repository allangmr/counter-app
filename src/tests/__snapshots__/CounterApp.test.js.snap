// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of CounterApp.js should be show <CounterApp /> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;

exports[`Test of CounterApp.js should be show <CounterApp /> with default value 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
